lib/common/pid.c

```diff
@@ -119,83 +137,111 @@ crm_read_pidfile(const char *filename)
     }
 
     if (read(fd, buf, sizeof(buf)) < 1) {
+        rc = errno;
         goto bail;
     }
 
-    if (sscanf(buf, "%ld", &pid) > 0) {
-        if (pid <= 0) {
-            pid = -ESRCH;
+    if (sscanf(buf, "%lld", &pid_read) > 0) {
+        if (pid_read <= 0) {
+            rc = ESRCH;
         } else {
-            crm_trace("Got pid %lu from %s\n", pid, filename);
+            rc = pcmk_rc_ok;
+            *pid = (pid_t) pid_read;
+            crm_trace("Read pid %lld from %s", pid_read, filename);
         }
     }
 
   bail:
-    if (fd >= 0) {
-        close(fd);
-    }
-    return pid;
+    close(fd);
+    return rc;
 }
 
-long
-crm_pidfile_inuse(const char *filename, long mypid, const char *daemon)
+/*!
+ * \internal
+ * \brief Check whether a process from a PID file matches expected values
+ *
+ * \param[in]  filename       Path of PID file
+ * \param[in]  expected_pid   If positive, compare to this PID
+ * \param[in]  expected_name  If not NULL, the PID from the PID file is valid
+ *                            only if it is active as a process with this name
+ * \param[out] pid            If not NULL, store PID found in PID file here
+ *
+ * \return Standard Pacemaker return code
+ */
+int
+pcmk__pidfile_matches(const char *filename, pid_t expected_pid,
+                      const char *expected_name, pid_t *pid)
 {
-    long pid = crm_read_pidfile(filename);
+    pid_t pidfile_pid = 0;
+    int rc = pcmk__read_pidfile(filename, &pidfile_pid);
 
-    if (pid < 2) {
-        // Invalid pid
-        pid = -ENOENT;
+    if (pid) {
+        *pid = pidfile_pid;
+    }
+
+    if (rc != pcmk_rc_ok) {
+        // Error reading PID file or invalid contents
         unlink(filename);
+        rc = ENOENT;
 
-    } else if (mypid && (pid == mypid)) {
-        // In use by us
-        pid = pcmk_ok;
+    } else if ((expected_pid > 0) && (pidfile_pid == expected_pid)) {
+        // PID in file matches what was expected
+        rc = pcmk_rc_ok;
 
-    } else if (crm_pid_active(pid, daemon) == FALSE) {
+    } else if (pcmk__pid_active(pidfile_pid, expected_name) == ESRCH) {
         // Contains a stale value
         unlink(filename);
-        pid = -ENOENT;
+        rc = ENOENT;
 
-    } else if (mypid && (pid != mypid)) {
+    } else if ((expected_pid > 0) && (pidfile_pid != expected_pid)) {
         // Locked by existing process
-        pid = -EEXIST;
+        rc = EEXIST;
     }
 
-    return pid;
+    return rc;
 }
 
+/*!
+ * \internal
+ * \brief
+ *
+ * \param[in] filename
+ * \param[in] name
+ *
+ * \return Standard Pacemaker return code
+ */
```

Comment:
Looks like you might have pasted a template here and forgotten to come back around to filling it in.

Reply:
Whoops :)